examples/higher-kinded.golden

```diff
@@ -0,0 +1,6 @@
+(of-unit #<IO action>) : (OfUnit IO)
+(of-unit #<closure>) : (OfUnit ((→) Unit))
```

Comment:
Is this a bug in the type pretty printer? I'd expect it to say `(OfUnit (→ Unit))` here.

Reply:
In core, the type `(-> A B C)` is printed as `A → (B → C)`, so I think other occurrences of `→` are intentionally printed as `(→)` in order to make sure `Monad (→) Unit` doesn't accidentally get printed as `Monad → Unit`, as that would be confusing.

When `→` has arguments, though, I agree it makes sense to print those arguments inside the parens.

Reply:
Thinking about this more, it's only the cases in which `→` has zero or one arguments which are a special case:

```
(→)
(→ A)
A → B
A → (B → C)
```

The current code accounts for the zero case, but not the one case.

Reply:
Fixed in e02824619e25f08ba2a2433ae38f185671218a84.

Reply:
Nice